vraptor-core/src/main/java/br/com/caelum/vraptor/serialization/gson/GsonBuilderWrapper.java

```diff
@@ -90,9 +91,10 @@ private void registerAdapter(Class<?> adapterType, Object adapter) {
 	}
 	
 	private Class<?> getAdapterType(Object adapter) {
-		Type[] genericInterfaces = adapter.getClass().getGenericInterfaces();
-		ParameterizedType type = (ParameterizedType) genericInterfaces[0];
-		Type actualType = type.getActualTypeArguments()[0];
+		final Class<?> klazz = extractRawTypeIfPossible(adapter.getClass());
+		final Type[] genericInterfaces = klazz.getGenericInterfaces();
+		final ParameterizedType type = (ParameterizedType) genericInterfaces[0];
+		final Type actualType = type.getActualTypeArguments()[0];
```

Comment:
It is a personal habit.
I think it better to understand the code. But I can take off if necessary.
Thank's.